README.md

```diff
@@ -66,8 +66,8 @@ gps-share is targetted specifically for Linux. It may or may not work on other
 POSIX hosts. Patches to add/fix support for non-Linux systems, are more than
 welcome.
 
-On Fedora, you might need to either disable selinux or configure it to allow
-gps-share to start and announce a TCP service.
+Remember to configure your firewall to allow your service to be reachable on the
+local network, as needed.
```

Comment:
> Since you know exactly what selinux policy to setup, I would like to specify that here. If I can't understand selinux, I wouldn't want others to have to figure out on their own either. :)

I hope to fix this upstream so it won’t be necessary. The problem affects all NMEA network sources that GeoClue can read from.

> Also "configure your firewall" is too generic, we should specify the avahi port at least.

“Configure your firewall” is just a reminder to check your configuration as part of troubleshooting if you need to adjust it.

> we should specify the avahi port at least.

As far as I know, Ubuntu and Fedora allows [m]DNS[/-SD] by default in FirewallD/UFW for their desktop variants. Many firewalls already have service definitions for mDNS, but documenting all of those is surely out of scope for this document.

> The service port of gps-share is typically dynamic (unless user specifies on commandline) so we can't be too specific about that in here.

I’ll address that in a separate bug report.

Reply:
>> 
>>    Since you know exactly what selinux policy to setup, I would like to specify that here. If I can't understand selinux, I wouldn't want others to have to figure out on their own either. :)
> 
> I hope to fix this upstream so it won’t be necessary. The problem affects all NMEA network sources that GeoClue can read from.

If by upstream, you mean in geoclue, that's a great idea. Firstly the NMEA source will likely be on a different machine than geoclue itself and geoclue itself doesn't need those rules. Secondly the protocol has been kept simple and generic so there could be other clients than geoclue making use of this and they'll need this info too. Best keep that info in docs of the actual software the needs them.



>>    Also "configure your firewall" is too generic, we should specify the avahi port at least.
>> 
> “Configure your firewall” is just a reminder to check your configuration as part of troubleshooting if you need to adjust it.

I meant if we can be easily specific, we should but i see yours and @hadess' point.

Reply:
>  If by upstream, you mean in geoclue, that's not a great idea.

gps-share doesn’t need the policy as it’s running unconstrained. gps-share is working just fine. GeoClue can’t read from any nmea-0183 network services without that policy change, however. (Including GeoClueShare and whatever other services implement the protocol.) This is [a bug](https://bugzilla.redhat.com/show_bug.cgi?id=1362118) in GeoClue’s policy.

Reply:
Ah ok, i confused the typical firewall issue with selinux.